llvm/lib/CodeGen/AsmPrinter/DwarfFile.cpp

```diff
@@ -27,8 +29,20 @@ void DwarfFile::addUnit(std::unique_ptr<DwarfCompileUnit> U) {
 // Emit the various dwarf units to the unit section USection with
 // the abbreviations going into ASection.
 void DwarfFile::emitUnits(bool UseOffsets) {
-  for (const auto &TheU : CUs)
+  bool FirstContribution = true;
+  for (const auto &TheU : CUs) {
+    getOrCreateAbbrevSectionStart();
+    TheU.get()->setAbbrevSectionBegin(AbbrevSectionStart);
+    // If there is only one Abbreviation contribution or we are working with the
+    // first compile unit, set the Abbreviation Offset label to be the same as
```

Comment:
first compile unit, we can avoid label arithmetic by setting the offset to the start of the section.